src/ConnectedMode.UnitTests/Binding/BindingProcessImplTests.cs

```diff
@@ -156,190 +111,20 @@ private static ServerExclusions CreateSettings()
-                .Callback<FixedStepsProgress>(x => ((IProgressStepExecutionEvents)notifications).ProgressChanged(x.Message, x.CurrentStep));
-
-            // Act
-            var result = await testSubject.DownloadQualityProfileAsync(progressAdapter.Object, CancellationToken.None);
-
-            // Assert
-            result.Should().BeTrue();
-            testSubject.InternalState.BindingConfigs.Should().NotContainKey(Language.Cpp);
-            testSubject.InternalState.BindingConfigs.Should().NotContainKey(Language.Secrets);
-            testSubject.InternalState.BindingConfigs.Should().NotContainKey(Language.Css);
-
-            testSubject.InternalState.BindingConfigs.Should().ContainKey(Language.CSharp);
-            testSubject.InternalState.BindingConfigs.Should().ContainKey(Language.VBNET);
-
-            // Progess notifications - percentage complete and messages
-            notifications.AssertProgress(1.0, 2.0, 3.0, 4.0);
-            CheckProgressMessages(languagesToBind);
-
-            // Check output messages
-            var missingPluginMessageCFamily = string.Format(Strings.SubTextPaddingFormat,
-                string.Format(BindingStrings.CannotDownloadQualityProfileForLanguage, Language.Cpp.Name));
-            var missingPluginMessageSecrets = string.Format(Strings.SubTextPaddingFormat,
-                string.Format(BindingStrings.CannotDownloadQualityProfileForLanguage, Language.Secrets.Name));
-
-            logger.AssertOutputStringExists(missingPluginMessageCFamily);
-            logger.AssertOutputStringExists(missingPluginMessageSecrets);
-            logger.AssertPartialOutputStringDoesNotExist(Language.Css.Name);
-
-            void CheckProgressMessages(params Language[] languages)
-            {
-                var expected = languages.Select(GetDownloadProgressMessages).ToArray();
-                notifications.AssertProgressMessages(expected);
-            }
-
-            static string GetDownloadProgressMessages(Language language)
-                => string.Format(BindingStrings.DownloadingQualityProfileProgressMessage, language.Name);
-        }
-
-        private void SetupAvailableLanguages(Mock<IBindingConfigProvider> configProvider,
-            Mock<ISonarQubeService> sonarQubeService,
-            params Language[] languages)
-        {
-            foreach (var language in languages)
-            {
-                var profile = ConfigureQualityProfile(sonarQubeService, language, "Profile" + language.Name);
-                configProvider.Setup(x => x.GetConfigurationAsync(profile, language, It.IsAny<BindingConfiguration>(), CancellationToken.None))
-                    .ReturnsAsync(Mock.Of<IBindingConfig>());
-            }
-        }
-
-        [TestMethod]
-        public async Task DownloadQualityProfile_WhenBindingConfigIsNull_Fails()
+        public async Task DownloadQualityProfile_CreatesBoundProjectAndCallsQPDownloader()
```

Comment:
Single new simple test that the downloader is called.

I moved the other tests from to the new "downloader" test class and then fixed them up.